torchrl/envs/gym_like.py

```diff
@@ -77,6 +77,7 @@ class GymLikeEnv(_EnvWrapper):
 
     @classmethod
     def __new__(cls, *args, **kwargs):
+        cls.batch_locked = True
```

Comment:
careful here: you're setting this in `__new__` then call super, which I think will override what you just wrote.
I would put a kwarg in the `__new__` of the parent class which defaults to True and for all the children pass that value upward.

Reply:
I thought it was better to set it in the init so that it's persistent if we change the default of EnvBase at some point